packages/babel-traverse/src/scope/index.ts

```diff
@@ -705,6 +705,8 @@ export default class Scope {
       for (const declar of declarations) {
         this.registerBinding(path.node.kind, declar);
       }
+    } else if (path.isClassDeclaration() && path.node.declare) {
+      this.registerBinding("class", path);
     } else if (path.isClassDeclaration()) {
       this.registerBinding("let", path);
```

Comment:
I think we should skip registering ambient class declaration here.

BTW we don't have a `class` binding type: https://github.com/babel/babel/blob/029cd15bd95b70e1164ac4d05ac2a4c178be5769/packages/babel-traverse/src/scope/binding.ts#L5-L13

Reply:
Sorry for declaraing an unsual binding type.
Can we just skip the registerDeclaration for declare class using `if(path.node.declare) return` inside binding block statement of registerDeclaration for class declaration?